spring-cloud-gcp-starters/spring-cloud-gcp-starter-pubsub/src/main/java/org/springframework/cloud/gcp/pubsub/autoconfig/GcpPubsubAutoConfiguration.java

```diff
@@ -112,4 +117,16 @@ public PublisherFactory defaultPublisherFactory(GcpProjectIdProvider projectIdPr
 		return new DefaultPublisherFactory(projectIdProvider, subscriberProvider, channelProvider,
 				credentialsProvider);
 	}
+
+	@Bean
+	@ConditionalOnMissingBean
+	public PubsubAdmin pubsubAdmin(GcpProjectIdProvider projectIdProvider) {
```

Comment:
Sorry, not sure what you mean with the autoconfig is missed?

I removed `@IntegrationComponentScan` and `@ComponentScan` and `PubsubAdmin` is correctly autowired. You were right that I didn't need them, they must had been there from previous experiments...